### PR TITLE
Button mixin

### PR DESCRIPTION
**Pull request by mturnwall**
The button mixin is now used to create button variations.

You get output like this now. We can tweak the output as needed.

```css
.btn-primary {
  background-color: rgba(75, 157, 249, 1);
  color: orange;
}

.btn-primary:hover {
  background-color: red;
}

.btn-primary:active, .btn-primary.active {
  background-color: red;
}

.btn-primary:focus {
  background-color: red;
}
```

### CODE REVIEW

File: styles/mixins/button.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 module.exports = function () {
   return function (mixin, bgColor, color) {
-    return {
-      'background-color': bgColor,
-      color,
-      display: 'inline-block',
-      'line-height': 1,
-      'text-align': 'center'
+    const activeBgColor = 'red';
```
Comment (jgallen23) [inline styles/mixins/button.js]: eh? red?

File: conf/default-colors.yaml [hunk 2]
```diff
@@ -8,3 +8,7 @@ color:
   link-hover: '{{color.secondary}}'
   input-background: '#fff'
   input-border: '#333'
+  button-primary-bg: 'rgba(75, 157, 249, 1)'
```
Comment (jgallen23) [inline conf/default-colors.yaml]: this should be the accent

File: conf/default-colors.yaml [hunk 3]
```diff
@@ -8,3 +8,7 @@ color:
...
+  button-primary-color: '{{color.accent}}'
```
Comment (jgallen23) [inline conf/default-colors.yaml]: this should be white (typically)

File: styles/mixins/button.js (hunk 1 above)
Comment (mturnwall) [inline styles/mixins/button.js]: I didn't know what colors to use so I just picked colors to test with.

Comment (jgallen23) [inline styles/mixins/button.js]: we should never have colors defined in the mixin. should pull in from config or passed in

Comment (mturnwall) [inline styles/mixins/button.js]: It's just a placeholder for testing. I mentioned this in the room but not sure anyone saw it. I'd like to add https://www.npmjs.com/package/postcss-colour-functions. We can use it for the hover effects on buttons. So `activeBgColor` would be `activeBgColor = darken(bgColor)`

Comment (dawnerd) [inline styles/mixins/button.js]: I think hover colors need to be defined in colors.yaml

Comment (jgallen23) [inline styles/mixins/button.js]: If we add support for those functions,we could still do darker and lighter in yaml

**Pull request merged**